src/components/autocomplete/js/autocompleteController.js

```diff
@@ -84,11 +84,18 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
         left   = hrect.left - root.left,
         width  = hrect.width,
         offset = getVerticalOffset(),
-        styles = {
-          left:     left + 'px',
-          minWidth: width + 'px',
-          maxWidth: Math.max(hrect.right - root.left, root.right - hrect.left) - MENU_PADDING + 'px'
-        };
+        styles;
+    // If using `md-input-container` layout, push dropdown up to cover `ng-messages` area
+    if ($attrs.mdFloatingLabel) {
+      left += 2;
+      top -= 18;
+      width -= 4;
+    }
```

Comment:
I agree, but I'm not sure how else to handle this.  This is reliant on CSS variables that are not available to JS.  Should I add a comment explaining further where these numbers came from?